src/main/java/ch/njol/skript/expressions/ExprAmount.java

```diff
@@ -77,12 +105,32 @@ public boolean isSingle() {
 	
 	@Override
 	public String toString(final @Nullable Event e, final boolean debug) {
-		return "amount of " + expr.toString(e, debug);
+		return (recursive ? "recursize size of " : "amount of ") + expr.toString(e, debug);
 	}
 	
+	@SuppressWarnings("unchecked")
 	@Override
 	protected Integer[] get(final Event e) {
+		if (recursive) {
+			boolean local = exprString.charAt(1) == '_';
+			// Local Variables: {_x}, Global Variables: {x}, substr will contain the value of x 
+			String substr = exprString.substring((local ? 2 : 1), exprString.length() - 1);
+			assert substr != null;
+			Object var = Variables.getVariable(substr, e, local); // We have to expose the entire variable with sublists included 
+			if (var != null)
+				return new Integer[] {getRecursiveSize((TreeMap<String, Object>) var)}; // Should already be a TreeMap 
+		}
 		return new Integer[] {expr.getArray(e).length};
 	}
 	
+	@SuppressWarnings("unchecked")
+	private static int getRecursiveSize(TreeMap<String, Object> map) {
+		int count = 0;
+		for (Map.Entry<String, Object> entry : map.entrySet()) {
```

Comment:
`*` index is actually a value, do you want to be counting that?

Reply:
Sorry, I'm not too sure what you mean.
I added a sysout (``System.out.println(entry.getKey() + "," + entry.getValue().getClass().getCanonicalName());``), and got this as output:
```
1,java.lang.String
2,java.util.TreeMap
null,java.lang.String
1,java.lang.String
2,java.lang.String
4
```
For the script:
```
command /test:
	trigger:
		set {_t::1} to "a"
		set {_t::2} to "b"
		set {_t::2::*} to "c"
		set {_t::2::1} to "d"
		set {_t::2::2} to "e"
		broadcast "%recursive size of {_t::*}%"
```

Reply:
nevermind, then. must have been some mundosk specific weirdness